### PR TITLE
update deps

### PR DESCRIPTION
**Pull request by 0xForerunner**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,19 +7,20 @@ edition = "2021"
...
+opentelemetry-http = "0.10"
+opentelemetry_sdk = { version = "0.21.0", features = ["rt-tokio"] }
 serde = "1.0.189"
 serde_json = "1.0.108"
 thiserror = "1.0.50"
 tokio = "1.33.0"
 tracing = "0.1.40"
 tracing-appender = "0.2.2"
-tracing-opentelemetry = "0.21.0"
+tracing-opentelemetry = "0.22"
```
Comment (0xKitsune) [inline Cargo.toml:23]: Let stick with precise versioning to avoid any possible risk of [accidential semver breaks or crates that dont adhere to semver](https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277/9).

Comment (0xForerunner) [reply Cargo.toml:23]: When you say precise, do you mean
```
http = "0.2.9" # (which is equivalent to http = "^0.2.9")
```
or
```
http = "=0.2.9" 
```
Because the latter is the only one that is resistant to semver breaks. The former only enforces a minimum version, not a maximum. Although it can be useful because you don't need to test for minimum version compatibility (Something that's doable in CI and increases the compatibility of your crate)

Comment (0xForerunner) [reply Cargo.toml:23]: Here's the relevent command to test minimal versions
```
cargo +nightly -Z minimal-versions update
```

Comment (0xKitsune) [reply Cargo.toml:23]: Good point, `0.2.9` still can experience semver breaks. Even still, patch versions strictly specify a minimum version. If using `0.2`, in the event that another package uses an outdated version that contains a bug (ex. `0.2.1` when latest is `0.2.2`), our program can compile with the outdated version.

With that being said, for well maintained libs, I'm fine with omitting patch versions.

**Review by 0xKitsune — APPROVED** [review]
